src/app/counter/counter-booking/counter-booking.component.html

```diff
@@ -164,8 +164,23 @@ <h2 class="text-center">Book Ticket</h2>
   </tbody>
 </table>
 <ngb-pagination
-  *ngIf="flights && flights.length"
-  [collectionSize]="flights ? flights.length : 0"
+  *ngIf="
```

Comment:
Was this the pagination not adjusting for the changes after filter? 

I pass an object called filterMetaData which contains a value called length. This length value is changed by all the filters and then read by the pagination. This works too, but I figured I would point it out if you were curious.

Reply:
It wasn't adjusting because I was just using `flights.length` for `collectionSize`.

I noticed that. Your way is probably more performant since it doesn't do the pipes multiple times, but I decided I'd rather not define another variable.